### PR TITLE
[full-ci] Shared with me split

### PR DESCRIPTION
**Pull request by kulmann**
Copied over from https://github.com/owncloud/web/pull/5177 - needed to rebase and can't push to the origin of that PR.

## Description
The "Shared with me" view shows 2 different lists for pending and accepted shares. On request user can see the declined files (clicking "show declined shares).

## Motivation and Context
General UX improvement: Motivate the user to accept or decline the shares through better overview of the statuses and highlighted pending section. 

## How Has This Been Tested?
UI test scenario: Sharing multiple files through one testing account with another
- test case 1: pending shares, 3 or less pending shares --> simple list of pending shares
- test case 2: pending shares, more than 3 shares --> "show all", "show less" option
- test case 3: no pending shares--> no pending shares list shown
- test case 4: no accepted shares--> "You are currently not collaborating on other people's resources" message shown, "show declined shares" option appears at pending shares
- test case 5: accepted shares exist--> list with accepted shares and option to show declined shares appears
- test case 6: no declined shares--> "No declined files found" message appears
- test case 7 declined shares exist --> list with declined shares shown on click of "show declined shares"

## Types of changes
- [x] New feature (non-breaking change which adds functionality)

## Checklist:
- [x] Code changes
- [x] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link> 

## Note
Please note that pagination has been removed for the shared with me view in this PR. UX needs a new concept here, as there are multiple tables on the page and the URL params for pagination cannot apply to all at the same time.

## General view

Screenshot 2021-09-21 at 16 16 48
Screenshot 2021-09-21 at 16 16 52

### CODE REVIEW

**Comment by ownclouders** [comment]
Results for <strong>oC10CreateDelete</strong> https://drone.owncloud.com/owncloud/web/19119/9/1
The following scenarios passed on retry:
- webUICreateFilesFolders/createFolders.feature:51

**Comment by ownclouders** [comment]
Results for <strong>oC10Files1</strong> https://drone.owncloud.com/owncloud/web/19119/12/1
The following scenarios passed on retry:
- webUIFilesActionMenu/versions.feature:15

**Comment by ownclouders** [comment]
Results for <strong>oC10SharingInternalUsersSharingIndicator</strong> https://drone.owncloud.com/owncloud/web/19130/28/1
The following scenarios passed on retry:
- webUISharingInternalUsersToRootSharingIndicator/shareWithUsers.feature:98

**Comment by ownclouders** [comment]
Results for <strong>oC10Files1</strong> https://drone.owncloud.com/owncloud/web/19130/12/1
The following scenarios passed on retry:
- webUIFilesActionMenu/versions.feature:15

**Comment by ownclouders** [comment]
Results for <strong>oCISSharingPublicExpireAndRoles</strong> https://drone.owncloud.com/owncloud/web/19133/62/1
The following scenarios passed on retry:
- webUISharingPublicExpire/shareByPublicLinkExpiringLinks.feature:24

**Comment by ownclouders** [comment]
Results for <strong>oC10SharingInternalGroupsSharingIndicator</strong> https://drone.owncloud.com/owncloud/web/19133/24/1
The following scenarios passed on retry:
- webUISharingInternalGroupsToRootSharingIndicator/shareWithGroups.feature:22

**Comment by ownclouders** [comment]
Results for <strong>oC10Files1</strong> https://drone.owncloud.com/owncloud/web/19133/12/1
The following scenarios passed on retry:
- webUIFilesActionMenu/versions.feature:15

**Comment by ownclouders** [comment]
Results for <strong>oC10CreateDelete</strong> https://drone.owncloud.com/owncloud/web/19139/9/1
The following scenarios passed on retry:
- webUIDeleteFilesFolders/deleteFilesFolders.feature:59

**Comment by ownclouders** [comment]
Results for <strong>oC10Files1</strong> https://drone.owncloud.com/owncloud/web/19139/12/1
The following scenarios passed on retry:
- webUIFilesActionMenu/versions.feature:36

**Comment by ownclouders** [comment]
Results for <strong>oC10SharingPublicManagement</strong> https://drone.owncloud.com/owncloud/web/19142/34/1
The following scenarios passed on retry:
- webUISharingPublicManagement/publicLinkIndicator.feature:110
- webUISharingPublicManagement/publicLinkIndicator.feature:27

File: packages/web-app-files/src/views/SharedWithMe.vue [hunk 1]
```diff
@@ -110,50 +181,120 @@ export default {
...
+          ...this.$route.params
+        },
+        query: {
+          ...this.$route.query,
+          'view-mode': this.sharesOtherViewMode
+        }
+      }
     },
 
+    // misc
```
Comment (pascalwengerter) [inline packages/web-app-files/src/views/SharedWithMe.vue:291]: Dev leftover?

**Review by pascalwengerter — APPROVED** [review]
LGTM 🏗️

File: packages/web-app-files/src/views/SharedWithMe.vue (hunk 1 above)
Comment (kulmann) [reply packages/web-app-files/src/views/SharedWithMe.vue:291]: nope, I tried to meta-group the computed props into props related to a) pending shares, b) accepted and declined shares and c) misc. ;-)

**Pull request merged**